cli/cmd/check.go

```diff
@@ -285,8 +254,14 @@ func renderInstallManifest(ctx context.Context) (*charts.Values, string, error)
 		return nil, "", err
 	}
 
+	// Ensure there is not already an existing Linkerd installation.
+	k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 30*time.Second)
+	if err != nil {
+		return values, "", err
+	}
```

Comment:
It looks like this is possibly an incomplete copy from `install.go` where we should be checking for the presence of the ConfigMap but we're not. This results in `linkerd check --pre` spewing out a lot more than just the current

```shell
$ linkerd check --pre
Can't install the Linkerd control plane in the 'linkerd' namespace. Reason: ConfigMap/linkerd-config already exists.
Run the command `linkerd upgrade`, if you are looking to upgrade Linkerd.
```

We need the rest of the check

```suggestion
	// Ensure there is not already an existing Linkerd installation.
	k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 30*time.Second)
	if err != nil {
		return values, "", err
	}
	// We just want to check if `linkerd-configmap` exists
	_, err := k8sAPI.CoreV1().ConfigMaps(controlPlaneNamespace).Get(cmd.Context(), k8s.ConfigConfigMapName, metav1.GetOptions{})
	if err == nil {
		fmt.Fprintf(os.Stderr, errMsgLinkerdConfigResourceConflict, controlPlaneNamespace, "ConfigMap/linkerd-config already exists")
		os.Exit(1)
	}
	if !kerrors.IsNotFound(err) {
		return err
	}
```